src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -205,6 +205,15 @@ internal void Start()
             gcTimer = t;
         }
 
+        internal void RemoveStuckGrainFromDirectory(ActivationData activationData)
+        {
+            DeactivateActivations(new List<ActivationData>() {activationData}).Ignore();
```

Comment:
I think a more appropriate here is DeactivateActivationOnIdle. 
We might need to even write a different method, if we find a need for a bit different sequence, but as of now  it is the same sequence. If you don't like the misleading name, lets rename DeactivateActivationOnIdle into DeactivateOneActivation